src/GraphQL.SystemReactive/SubscriptionExecutionStrategy.cs

```diff
@@ -1,178 +0,0 @@
-
-            ExecutionResult result = new SubscriptionExecutionResult
-            {
-                Executed = true,
-                Streams = streams
-            }.With(context);
-
-            return result;
-        }
-
-        private async Task<IDictionary<string, IObservable<ExecutionResult>>> ExecuteSubscriptionNodesAsync(ExecutionContext context, ExecutionNode[] nodes)
-        {
-            var streams = new Dictionary<string, IObservable<ExecutionResult>>();
-
-            foreach (var node in nodes)
-            {
-                var stream = await ResolveEventStreamAsync(context, node).ConfigureAwait(false);
-
-                if (stream != null)
-                    streams[node.Name!] = stream;
-            }
-
-            return streams;
-        }
-
-        /// <summary>
-        /// Asynchronously returns a stream of <see cref="ExecutionResult"/> responses for the
-        /// specified <see cref="ExecutionNode"/>.
-        /// </summary>
-        protected virtual async Task<IObservable<ExecutionResult>?> ResolveEventStreamAsync(ExecutionContext context, ExecutionNode node)
-        {
-            context.CancellationToken.ThrowIfCancellationRequested();
-
-            object? source = (node.Parent != null)
-                ? node.Parent.Result
-                : context.RootValue;
-
-            try
-            {
-                var resolveContext = new ResolveFieldContext(new ReadonlyResolveFieldContext(node, context));
-
-                IObservable<object?> subscription;
-
-                if (node.FieldDefinition?.Subscriber != null)
-                {
-                    subscription = await node.FieldDefinition.Subscriber.SubscribeAsync(resolveContext).ConfigureAwait(false);
-                }
-                else
-                {
-                    throw new InvalidOperationException($"Subscriber not set for field '{node.Field.Name}'.");
-                }
-
-                // The IServiceProvider instance will be disposed at this point, set it to null instead of exposing the disposed object
-                resolveContext.RequestServices = null;
-
-                return subscription
-                    .Select(value => BuildSubscriptionExecutionNode(node.Parent!, node.GraphType!, node.Field, node.FieldDefinition, node.IndexInParentNode, value!))
-                    .SelectMany(async executionNode =>
-                    {
-                        if (context.Listeners != null)
-                        {
-                            foreach (var listener in context.Listeners)
-                            {
-                                await listener.BeforeExecutionAsync(context)
-                                    .ConfigureAwait(false);
-                            }
-                        }
-
-                        // Execute the whole execution tree and return the result
-                        await ExecuteNodeTreeAsync(context, executionNode).ConfigureAwait(false);
```

Comment:
It's here that errors added within ExecuteNodeTreeAsync get caught and added to `context`, but never cleared until the subscription is disposed/disconnected.

Reply:
👍 Thanks.